### PR TITLE
p2p metrics database and queries

### PR DESCRIPTION
**Pull request by maackle**
- adds `p2p_metrics` db
- renames `p2p` db to `p2p_state` in various places
- adds one mutating and two reading SQL queries on the metrics db
- creates kitsune actor methods for reading and writing the metrics db, but those are only partially hooked up

The actor plumbing is all in place, but with some TODOs in the intermediate stages. I figure we can finalize the plumbing once we actually start to use the metrics API in kitsune.

### CODE REVIEW

File: crates/holochain_sqlite/src/db/p2p_metrics.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        .as_nanos()
+        .try_into()
+        .map_err(|e: TryFromIntError| DatabaseError::Other(e.into()))
+}
+
+fn time_from_ns(ns: u64) -> SystemTime {
+    std::time::UNIX_EPOCH
+        .checked_add(Duration::from_nanos(ns))
+        .expect("Time must be after 1970 and before the end of the universe")
+}
```
Comment (neonphog) [inline crates/holochain_sqlite/src/db/p2p_metrics.rs:27]: sqlite stores integers as `i64`s. Not that we're going to overflow time-wise, but might be good to pedantically make the types match.

Also, we've got other code storing timestamps as milliseconds, and yet other code storing our timestamps as binary blobs, we should standardize on something...

File: crates/holochain/src/conductor/handle.rs [hunk 2]
```diff
@@ -412,14 +416,49 @@ impl<DS: DnaStore + 'static> ConductorHandleT for ConductorHandleImpl<DS> {
...
             }
+            PutMetricDatum {
+                span_context,
+                respond,
+                dna_hash,
+                to_agent,
+                metric,
+            } => {
+                let env = { self.conductor.read().await.p2p_env(space) };
+                let agent = todo!("can we use to_agent?");
```
Comment (neonphog) [inline crates/holochain/src/conductor/handle.rs:427]: I don't think to_agent is correct... that's used mainly to dispatch to cells--meaning it's the local destination agent for events.

But, this call, and all the other p2p calls are intercepted before they get to the cell, they really shouldn't even *have* a to_agent, and sometimes they're filled in with garbage data...

Also, the type is wrong, we need something like `remote_agent: Arc<KitsuneAgent>`.

File: crates/holochain/src/conductor/handle.rs [hunk 3]
```diff
@@ -412,14 +416,49 @@ impl<DS: DnaStore + 'static> ConductorHandleT for ConductorHandleImpl<DS> {
...
+                let res = put_metric_datum(env, agent, metric, SystemTime::now())
+                    .map_err(holochain_p2p::HolochainP2pError::other);
```
Comment (neonphog) [inline crates/holochain/src/conductor/handle.rs:429]: I think I need direct control of the timestamp from the kitsune side, so if I start a sync at X, and the query to the remote side takes 2.3 seconds to succeed, I still mark the sync success time as X.

File: crates/holochain_sqlite/src/db/p2p_metrics.rs (hunk 1 above)
Comment (maackle) [reply crates/holochain_sqlite/src/db/p2p_metrics.rs:27]: Yeah... @freesig what do you think? `i64` nanoseconds will overflow in the year 2270, but gives total precision. Maybe we can split the difference and go with microseconds for a cool 300,000 years?

Comment (neonphog) [reply crates/holochain_sqlite/src/db/p2p_metrics.rs:27]: There are still processors out there that can't even return nanosecond precision times. I don't think we'd have any problems where micros weren't good enough.

**Comment by maackle** [comment]
A few compiler errors crept in here but are fixed in #808

**Review by neonphog — APPROVED** [review]

**Pull request merged**